src/VisualStudio/IntegrationTest/TestUtilities/VisualStudioInstanceFactory.cs

```diff
@@ -21,6 +21,9 @@ namespace Microsoft.VisualStudio.IntegrationTest.Utilities
 {
     public sealed class VisualStudioInstanceFactory : IDisposable
     {
+        [ThreadStatic]
+        private static bool s_inHandler;
```

Comment:
Isn't `ThreadLocal<T>` generally preferred (or is that only when you need to supply an initializer)?